easybuild/tools/module_generator.py

```diff
@@ -260,7 +268,18 @@ def prepend_paths(self, key, paths, allow_abs=False, expand_relpaths=True):
         paths = self._filter_paths(key, paths)
         if paths is None:
             return ''
-        return self.update_paths(key, paths, prepend=True, allow_abs=allow_abs, expand_relpaths=expand_relpaths)
+        if key == 'PYTHONPATH' and build_option('prefer_ebpythonprefix_over_pythonpath'):
+            # Special condition for PYTHONPATHs that match the standard pattern,
+            # replace with EBPYTHONPREFIX which is added to python sys path at runtime
+            python_paths = {path for path in paths if re.match(r'lib/python\d+\.\d+/site-packages', path)}
+            if python_paths:
+                self.log.info("Replaced PYTHONPATH %s with EBPYTHONPREFIX", python_paths)
+                paths = [path for path in paths if path not in python_paths]
+                ret = self._update_paths('EBPYTHONPREFIX', [''], prepend, allow_abs, expand_relpaths)
```

Comment:
As per your comment that there should be only a single such path:

```suggestion
            python_paths = [path for path in paths if re.match(r'lib/python\d+\.\d+/site-packages', path)]
            if python_paths:
                if len(python_paths) > 1:
                    raise EasyBuildError('Found multiple paths for PYTHONPATH: ' + ', '.join(pythonpaths))
                python_path = python_paths[0]
                self.log.info("Replaced PYTHONPATH %s with EBPYTHONPREFIX", python_path)
                ret = self._update_paths('EBPYTHONPREFIX', [''], prepend, allow_abs, expand_relpaths)
                paths = [path for path in paths if path != python_path]
```

I can't imagine a reason where multiple such paths would be valid, so that should be safer and makes e.g. the log message clearer ("Replaced PYTHONPATH lib/foo with EBPYTHONPREFIX" instead of "Replaced PYTHONPATH {/lib/foo} with EBPYTHONPREFIX")